node/core/pvf/common/src/worker/security.rs

```diff
@@ -35,18 +35,36 @@ pub mod landlock {
 
 	use landlock::{Access, AccessFs, Ruleset, RulesetAttr, RulesetError, ABI};
 
-	/// Version of landlock ABI. Use the latest version supported by our reference kernel version.
+	/// Landlock ABI version. Use the latest version supported by our reference kernel version.
+	///
+	/// # Versions (June 2023)
 	///
 	/// - Reference kernel version: 5.16+
 	/// - ABI V1: 5.13
 	/// - ABI V2: 5.19
 	///
-	/// Please update these docs if they are out-of-date, and bump the ABI version to the minimum
+	/// Please update these values if they are out-of-date, and bump the ABI version to the minimum
 	/// ABI supported by the reference kernel version.
+	///
+	/// # Determinism
+	///
+	/// You may wonder whether we could always use the latest ABI instead of the ABI supported by
+	/// the reference kernel version. It seems plausible, since landlock provides a best-effort
+	/// approach to enabling sandboxing. For example, if the reference version only supported V1 and
+	/// we were on V2, then landlock would use V2 if it was supported on the current machine, and
+	/// just fall back to V1 if not.
+	///
+	/// The issue with this is indeterminacy. If half of validators were on V2 and half were on V1,
```

Comment:
This logic doesn't parse well for me, if V1 is less restrictive and makes validators vulnerable then I would guess having just a part of them vulnerable is better than all of them isn't it ?

Reply:
Thank you for the explanation, makes sense now! I guess V1 is the best we can do.

Reply:
It should be the default for validators to have these security measures in place, ideally we would have none without them. Anyhow, the risk of disputes should be very low as this is already a second level defense mechanism. I would rather have a dispute than some PVF being able to read the validator's disk. We should make damn sure that there are no legitimate disk accesses of course, but checking that should be independent of PVF or candidate, so also rather easy. At least at the moment, I can't think of a legitimate disk access that would only happen on some PVFs ..

Reply:
Thanks @eskimor. Determinism is still a goal, and given that ABI v2 and v3 don't add to the security I would stick to v1 here.[^1] I will update the docs as the determinacy is still relevant but not the only factor. And if in the future a version is released with meaningful new features like network blocking (which is an eventual goal of landlock) we can enable it immediately. We should keep in mind that attackers can exploit any difference in operation to attack the chain, but the risk is low and there are other indeterminacy vectors anyway.

[^1]: v2 adds a new config option which we don't use, and v3 additionally blocks file truncation - which might be annoying, but not really critical to security, and validators should have backups, right?